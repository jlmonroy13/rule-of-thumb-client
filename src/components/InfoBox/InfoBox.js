import React from 'react';
import PropTypes from 'prop-types';

const InfoBox = ({ title, subtitle, message, onClose }) => (
  <div className="info-box" data-testid="info-box">
    <div className="info-box__container">
      <div className="mr-5">
        <span className="info-box__subtitle">{subtitle}</span>
        <span className="info-box__title">{title}</span>
      </div>
      <p className="info-box__message">{message}</p>
    </div>
    <button
      className="info-box__button"
      data-testid="info-box-close-button"
      onClick={onClose}
    >
      &#x2715;
    </button>
  </div>
);

InfoBox.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  onClose: PropTypes.func.isRequired,
};

export default InfoBox;
