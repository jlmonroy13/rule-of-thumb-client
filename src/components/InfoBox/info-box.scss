.info-box {
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: $gallery;
  padding: 20px 25px;
  margin-bottom: 30px;

  @include media-breakpoint-down(sm) {
    align-items: flex-start;
  }

  &__container {
    display: flex;
    align-items: center;

    @include media-breakpoint-down(xl) {
      flex-direction: column;
      align-items: flex-start;
    }
  }

  &__message {
    @include font-size(15);

    letter-spacing: 0;
    line-height: 23px;
    font-weight: 300;
    max-width: 733px;
    margin: 0;
  }

  &__title {
    @include font-size(37);

    line-height: 0.8;
    font-weight: 700;
    display: block;

    @include media-breakpoint-down(xl) {
      margin-bottom: 10px;
    }
  }

  &__subtitle {
    @include font-size(22);

    line-height: 0.8;
    font-weight: 300;
  }

  &__button {
    @include font-size(20);
  }
}
