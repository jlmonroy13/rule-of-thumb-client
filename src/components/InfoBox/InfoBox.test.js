/**
 *
 * Tests for InfoBox
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';

import InfoBox from './index';

const props = {};

describe('<InfoBox />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<InfoBox {...props} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<InfoBox {...props} />);
    expect(firstChild).toMatchSnapshot();
  });
});
