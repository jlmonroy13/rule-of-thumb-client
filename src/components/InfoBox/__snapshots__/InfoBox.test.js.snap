// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoBox /> Should render and match the snapshot 1`] = `
<div
  class="info-box"
>
  <div>
    <span>
      Speak out. Be heard
    </span>
    <b>
      Be counted
    </b>
  </div>
  <div>
    Rule of Thumb is a crowd sourced court of public opinion where anyone and everyone can speak out and speak freely. It's easy: You share your opinion, we analyze and put the data in a public report.
  </div>
  <button>
    âœ•
  </button>
</div>
`;
